### PR TITLE
fix various discrepancies for watcom

### PR DESCRIPTION
**Pull request by jmalak**

### CODE REVIEW

File: src/lang.c [hunk 1]
```diff
@@ -7,11 +7,17 @@
 
 #define yy_create_buffer _w32__create_buffer
 #define yy_delete_buffer _w32__delete_buffer
-#define yy_flex_debug _w32__flex_debug
+#define yy_scan_buffer _w32__scan_buffer
```
Comment (gvanem) [inline src/lang.c:10]: `lang.c` is generated from `lang.l`. So no point patching that.
You could instead fix Wmake to regenerate `lang.c` (using flex) when it should. 

Now when I do `del lang.c & wmake -f watcom_w.mak`, it stupidly complains:
```
language.c(50): Error! E1055: Unable to open 'lang.c'
```

Comment (jmalak) [reply src/lang.c:10]: OK, I will fix it. I supposed it is maintenance task because lang.c is in repository.

Comment (jmalak) [reply src/lang.c:10]: Do you know where can I download latest flex for Windows?

**Comment by jmalak** [comment]
it is tested on linux, I don't get any up-to-date flex version for Windows

File: src/lang.c (hunk 1 above)
Comment (gvanem) [reply src/lang.c:10]: funny-google-search-animated-gif-1

**Comment by gvanem** [comment]
I use flex as bundled from CygWin32. But here is a Win-Flex:
https://sourceforge.net/projects/winflexbison/

**Comment by gvanem** [comment]
I will have to apply these changes by hand since I've modified many of the same files.
Tomorrow maybe?

**Comment by jmalak** [comment]
I am trying to compile application with build libraries and have one problem with select function.
It looks like it doesn't exist, only for DJGPP (DJGPP DOS library has select) and Windows (Winsock).
Could you explain how select is handled in Watt-32.

**Comment by gvanem** [comment]
Not sure about `select()`. I've had the same problem with Watcom and it's calling convention.

Check a .map-file, `dumpbin /exports watt-32.dll` etc. But `select_s()` is in there.

**Comment by jmalak** [comment]
Yes, select_s exists, but by example bind.c call select(....) not select_s().
Should be mapped select to select_s ?

**Comment by gvanem** [comment]
Are you talking about `WIN32`? Check `select.c`:
```c
#if defined(WIN32)
int W32_CALL select (int nfds, fd_set *read_fds, fd_set *write_fds,
                     fd_set *except_fds, struct timeval *timeout)
{
  return select_s (nfds, read_fds, write_fds, except_fds, timeout);
}
#endif
```

**Comment by jmalak** [comment]
No, I am talking about DOS 16/32-bit.

**Comment by gvanem** [comment]
Then  `select()` should probably not be inside that `(WIN32)`.

File: src/makefile.all [hunk 2]
```diff
@@ -644,16 +644,16 @@ $(WATT_DLL): $(OBJS) $(RESOURCE) $(LINKARG)
 $(STAT_LIB): $(OBJS) $(LIBARG)
 	$(AR) -q -b -c -pa $^@ @$(LIBARG)
 
--!include "$(OBJROOT)watt32.dep"
```
Comment (gvanem) [inline src/makefile.all:647]: What happened to this file now?

File: src/punycode.c [hunk 3]
```diff
@@ -402,7 +403,7 @@ int main (int argc, char **argv)
       if (r == EOF || r == 0)
          break;
 
-      if (r != 2 || uplus[1] != '+' || codept > (DWORD) - 1)
```
Comment (gvanem) [inline src/punycode.c:405]: Why? What do you think could happen on Cygwin-x64 here?

**Comment by gvanem** [comment]
I've added most of these changes by hand.
And those Watcom `#pragma` statements.

File: src/makefile.all (hunk 2 above)
Comment (jmalak) [reply src/makefile.all:647]: Watcom wmake use dependency info included in .obj files by Watcom compilers therefore doesn't nee any dependency info in this file, It need not be used for watcom and wmake

File: src/punycode.c (hunk 3 above)
Comment (jmalak) [reply src/punycode.c:405]: Oh sorry 64-bit Windows exception, need to be fixed.

File: src/makefile.all (hunk 2 above)
Comment (gvanem) [reply src/makefile.all]: I see. Removed and pushed your stuff in this PR.
Hence closing. Thanks!

**Comment by gvanem** [comment]
Closing as explained.

**Pull request closed without merging**